### ISSUE TITLE
[docker] Fix the Dockerfile for latest code

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1244]
Hello @dbeltrankyl , @mcastril 

I was testing a build of Autosubmit using GitHub Actions, and added the exact line from the `dockerfiles/README.md`, that we sent to EDITO-Infra team around August last year.

The command failed on GitHub, and when I tried locally I got the exact same error. Might be something easy, but we better fix it sooner rather than later (before we have an upcoming deadline for EDITO).

```bash
kinow@ranma:~/Development/python/workspace/autosubmit/dockerfiles$ docker build \
            -t ${USER}/autosubmit:4.0.84-bullseye-slim \
            -t ${USER}/autosubmit:latest \
            .
[+] Building 28.0s (20/25)                                       docker:default
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 6.59kB                                     0.0s
 => [internal] load metadata for docker.io/library/debian:bullseye-slim    2.1s
 => [internal] load metadata for docker.io/mambaorg/micromamba:1.4.9-bull  2.2s
 => [auth] mambaorg/micromamba:pull token for registry-1.docker.io         0.0s
 => [auth] library/debian:pull token for registry-1.docker.io              0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] preparing inline document                                   0.0s
 => [micromamba 1/1] FROM docker.io/mambaorg/micromamba:1.4.9-bullseye-s  17.2s
 => => resolve docker.io/mambaorg/micromamba:1.4.9-bullseye-slim@sha256:1  0.0s
 => => sha256:2083d5e3acfffcaca91475053252e8f3816a55e4400 2.37kB / 2.37kB  0.0s
 => => sha256:389c26cacad934fc7dd8fcfd259f44791ce175a9cb4 5.74kB / 5.74kB  0.0s
 => => sha256:1b459e9c6968a32ae5a4321a1f14dfbb8e9f1cb57d8 2.36kB / 2.36kB  0.0s
 => => sha256:072e55e492c8f53448eed58aa077e344eccc7b4 115.22kB / 115.22kB  1.3s
 => => sha256:14726c8f78342865030f97a8d3492e2d1a68fbd2 31.42MB / 31.42MB  14.7s
 => => sha256:2712dc90660a8ae3d5fea88bd5afd6e1063a9a66668 5.70MB / 5.70MB  7.6s
 => => sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d 32B / 32B  8.1s
 => => sha256:3ff1700158b11b432c1646cdcb66e7c2170a58a17bf7ef0 524B / 524B  8.9s
 => => sha256:492af156d32e36bc874a7fda25517dd9cf7477f35b50939 285B / 285B  9.4s
 => => sha256:f44123b607d41a6ff5e3e2934fcb8f3b6121750c95 4.55kB / 4.55kB  10.1s
 => => sha256:bf9cfdb98097403328f35cda87e4956b1e9c748a5de09c 683B / 683B  10.7s
 => => sha256:87bcfcd4e3d942fd007f334bb07c32b0147ad0f5c5bb2f 641B / 641B  11.2s
 => => sha256:fc3edf513487fc003623af5b960c953f3f2e9ffd4a43aa 345B / 345B  11.5s
 => => extracting sha256:14726c8f78342865030f97a8d3492e2d1a68fbd22778f9a3  1.8s
 => => extracting sha256:072e55e492c8f53448eed58aa077e344eccc7b40c5ea9c70  0.0s
 => => extracting sha256:2712dc90660a8ae3d5fea88bd5afd6e1063a9a666685ca3b  0.2s
 => => extracting sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6  0.0s
 => => extracting sha256:3ff1700158b11b432c1646cdcb66e7c2170a58a17bf7ef0c  0.0s
 => => extracting sha256:492af156d32e36bc874a7fda25517dd9cf7477f35b509392  0.0s
 => => extracting sha256:f44123b607d41a6ff5e3e2934fcb8f3b6121750c954020a7  0.0s
 => => extracting sha256:bf9cfdb98097403328f35cda87e4956b1e9c748a5de09c16  0.0s
 => => extracting sha256:87bcfcd4e3d942fd007f334bb07c32b0147ad0f5c5bb2ff1  0.0s
 => => extracting sha256:fc3edf513487fc003623af5b960c953f3f2e9ffd4a43aaab  0.0s
 => [stage-1  1/17] FROM docker.io/library/debian:bullseye-slim@sha256:c  17.6s
 => => resolve docker.io/library/debian:bullseye-slim@sha256:c6d9e246479d  0.0s
 => => sha256:c6d9e246479d56687c1a579a7a0336956a5ce6f2bc2 1.85kB / 1.85kB  0.0s
 => => sha256:49caa2d0610fd478d9aa58fa9dd449d7d76a201bc502a00 529B / 529B  0.0s
 => => sha256:71f8bbcd77b8db961035cc16424629f1f7482fdf3fe 1.46kB / 1.46kB  0.0s
 => => sha256:5d0aeceef7eeb53c3f853fb229ea7fd13a5a56f4 31.42MB / 31.42MB  15.6s
 => => extracting sha256:5d0aeceef7eeb53c3f853fb229ea7fd13a5a56f4ba371ca4  1.7s
 => [stage-1  2/17] COPY --from=micromamba /bin/micromamba /bin/micromamb  0.1s
 => [stage-1  3/17] COPY --from=micromamba /usr/local/bin/_activate_curre  0.0s
 => [stage-1  4/17] COPY --from=micromamba /usr/local/bin/_dockerfile_she  0.0s
 => [stage-1  5/17] COPY --from=micromamba /usr/local/bin/_entrypoint.sh   0.0s
 => [stage-1  6/17] COPY --from=micromamba /usr/local/bin/_dockerfile_ini  0.0s
 => [stage-1  7/17] COPY --from=micromamba /usr/local/bin/_dockerfile_set  0.0s
 => [stage-1  8/17] RUN /usr/local/bin/_dockerfile_initialize_user_accoun  0.4s
 => [stage-1  9/17] RUN if grep -q '^ID=alpine$' /etc/os-release; then     0.4s
 => [stage-1 10/17] WORKDIR /app/autosubmit/                               0.1s
 => [stage-1 11/17] RUN mkdir -pv "/app/autosubmit//logs" &&     mkdir -p  0.3s
 => ERROR [stage-1 12/17] RUN apt update &&     apt install -y       bash  6.2s
------
 > [stage-1 12/17] RUN apt update &&     apt install -y       bash=5.1-2+deb11u1       ca-certificates=20210119       curl=7.74.0-1.3+deb11u7       desktop-file-utils=0.26-1       dialog=1.3-20201126-1       graphviz=2.42.2-5       iputils-ping=3:20210202-1       less=551-2       net-tools=1.60+git20181103.0eebece-1       openssh-server=1:8.4p1-5+deb11u1       python3-tk=3.9.2-1       sqlite3=3.34.1-3       sudo=1.9.5p2-3+deb11u1       vim=2:8.2.2434-3+deb11u1       xdg-utils=1.1.3-4.1 &&     update-ca-certificates &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*:
0.380 
0.380 WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
0.380 
0.482 Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
0.600 Get:2 http://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
0.639 Get:3 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
0.713 Get:4 http://deb.debian.org/debian bullseye/main amd64 Packages [8068 kB]
3.204 Get:5 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [269 kB]
3.247 Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [18.8 kB]
4.330 Fetched 8564 kB in 4s (2181 kB/s)
4.330 Reading package lists...
5.011 Building dependency tree...
5.152 Reading state information...
5.173 All packages are up to date.
5.183 
5.183 WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
5.183 
5.190 Reading package lists...
5.823 Building dependency tree...
5.963 Reading state information...
6.111 E: Version '7.74.0-1.3+deb11u7' for 'curl' was not found
6.111 E: Version '1:8.4p1-5+deb11u1' for 'openssh-server' was not found
------
Dockerfile:96
--------------------
  95 |     # TODO: add something like xpdf=3.04+git20210103-3 if GUI/X is needed.
  96 | >>> RUN apt update && \
  97 | >>>     apt install -y \
  98 | >>>       bash=5.1-2+deb11u1 \
  99 | >>>       ca-certificates=20210119 \
 100 | >>>       curl=7.74.0-1.3+deb11u7 \
 101 | >>>       desktop-file-utils=0.26-1 \
 102 | >>>       dialog=1.3-20201126-1 \
 103 | >>>       graphviz=2.42.2-5 \
 104 | >>>       iputils-ping=3:20210202-1 \
 105 | >>>       less=551-2 \
 106 | >>>       net-tools=1.60+git20181103.0eebece-1 \
 107 | >>>       openssh-server=1:8.4p1-5+deb11u1 \
 108 | >>>       python3-tk=3.9.2-1 \
 109 | >>>       sqlite3=3.34.1-3 \
 110 | >>>       sudo=1.9.5p2-3+deb11u1 \
 111 | >>>       vim=2:8.2.2434-3+deb11u1 \
 112 | >>>       xdg-utils=1.1.3-4.1 && \
 113 | >>>     update-ca-certificates && \
 114 | >>>     apt-get clean && \
 115 | >>>     rm -rf /var/lib/apt/lists/*
 116 |     
--------------------
ERROR: failed to solve: process "/usr/local/bin/_dockerfile_shell.sh apt update &&     apt install -y       bash=5.1-2+deb11u1       ca-certificates=20210119       curl=7.74.0-1.3+deb11u7       desktop-file-utils=0.26-1       dialog=1.3-20201126-1       graphviz=2.42.2-5       iputils-ping=3:20210202-1       less=551-2       net-tools=1.60+git20181103.0eebece-1       openssh-server=1:8.4p1-5+deb11u1       python3-tk=3.9.2-1       sqlite3=3.34.1-3       sudo=1.9.5p2-3+deb11u1       vim=2:8.2.2434-3+deb11u1       xdg-utils=1.1.3-4.1 &&     update-ca-certificates &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100

```

**Comment by kinow** [comment]
mentioned in issue requests#2370

**Comment by kinow** [comment]
These were fixed in 4.1.7 when we had to deploy it for EDITO.
